### DISCUSSION TITLE
Private storage: How to hook into the plugin to generate the temporary url?

### DISCUSSION THREAD

**Discussion by willcastillo** [awcodes/filament-tiptap-editor#443] [Q&A]
Hello,

The plugin is working as expected... except that it is showing the base url in the editor for a resource that is hosted in a private disk, instead of its temporary url.

In other words, it is storing the url as it should, only the base url, so we can update it at runtime... however, I'm not sure how to do it.

Any hint on how to achieve this? The only way I can think of is using `afterStateHydrated` and parsing every single url... but not sure if there's a better way to do it.

**Comment by awcodes** [comment]
Private files don't really work well in WYSIWYG editors. You would have to parse the content every time it is read back out, because the token for the private disk would have expired by then.

**Reply by willcastillo** [reply]
Thanks!
